src/rdsn/src/meta/meta_backup_engine.cpp

```diff
@@ -179,6 +179,86 @@ void meta_backup_engine::backup_app_partition(const gpid &pid)
         });
 }
 
+// ThreadPool: THREAD_POOL_DEFAULT
+void meta_backup_engine::on_backup_reply(const error_code err,
+                                         const backup_response &response,
+                                         const gpid &pid,
+                                         const rpc_address &primary)
+{
+    {
+        zauto_read_lock l(_lock);
+        if (_is_backup_failed) {
```

Comment:
Is there a status of BACKUP_FAILED?  Why do we still need a flag variable?

Reply:
`on_backup_reply` only handle one partition backup, `_is_backup_failed` will set true when one partition backup failed, all partitions needs to be failed, you can reference function `handle_replica_backup_failed`.